src/Symfony/Component/Console/Cursor.php

```diff
@@ -21,8 +21,16 @@ final class Cursor
     private $output;
     private $input;
 
-    public function __construct(OutputInterface $output, $input = STDIN)
+    public function __construct(OutputInterface $output, $input = null)
     {
+        if (null === $input && !\defined('STDIN')) {
+            throw new \TypeError(sprintf('The second argument of "%s()" must be a resource.', __METHOD__));
```

Comment:
instead of the exception, what about this?:
`$input = $input ?? (\defined('STDIN') ? STDIN : fopen('php://input', 'r'));`

Reply:
I thought about that to, but I wasn't sure if using `php://input` will always work or whether throwing would be more useful. What do you think?

Reply:
AFAIK, `php://input` always works.

Reply:
let's use your suggestion then